examples/fuse_examples/multimodality/ehr_transformer/dataset.py

```diff
@@ -0,0 +1,434 @@
+import pickle
+from collections import OrderedDict
+from random import randrange
+from typing import Sequence, Any
+import re
+import glob
+import os
+
+import numpy as np
+import pandas as pd
+from typing import Tuple
+
+from fuse.data import DatasetDefault, PipelineDefault, OpBase
+from fuse.data.ops.ops_read import OpReadDataframe
+
+# from fuse.data.ops.ops_common import OpCond, OpSet
+from fuse.data.utils.split import dataset_balanced_division_to_folds
+from ops_read_cinc import OpReadDataframeCinC
+from fuse.data.utils.export import ExportDataset
+
+SOURCE = r"C:/D_Drive/Projects/EHR_Transformer/PhysioNet/predicting-mortality-of-icu-patients-the-physionetcomputing-in-cardiology-challenge-2012-1.0.0/predicting-mortality-of-icu-patients-the-physionet-computing-in-cardiology-challenge-2012-1.0.0"
+
+VALID_TESTS_ABOVE_ZERO = [
+    "pH",
+    "Weight",
+    "Height",
+    "DiasABP",
+    "HR",
+    "NIMAP",
+    "MAP",
+    "NIDiasABP",
+    "NISysABP",
+    "PaCO2",
+    "PaO2",
+    "Temp",
+    "SaO2",
+    "RespRate",
+    "SysABP",
+]
+
+STATIC_FIELDS = ["Age", "Gender", "Height", "ICUType", "Weight"]
+
+
+class OpAddBMI(OpBase):
+    def __call__(self, sample_dict) -> Any:
+
+        d_static = sample_dict["StaticDetails"]
+
+        if ("Height" in d_static.keys()) & ("Weight" in d_static.keys()):
```

Comment:
I prefer not to add BMI with -1 to the dict of static values in case it can't be calculated from missing Height or Weight.
It will be further converted to nan during percentile calculations and will be skipped in the next steps of trajectories bild.